File: src/ollama_client.py
import requests
from config import config


def generate_agent_reply(messages, max_tokens=250):
    endpoint = f"{config['ollama_api_url'].rstrip('/')}/v1/generate"
    payload = {
        "model": config["ollama_model"],
        "prompt": messages,
        "max_tokens": max_tokens,
        "temperature": config["ollama_temperature"],
    }

    response = requests.post(endpoint, json=payload)
    if not response.ok:
        raise RuntimeError(f"Ollama API error {response.status_code}: {response.text}")

    data = response.json()
    if isinstance(data.get("response"), str):
        return data["response"].strip()

    if isinstance(data.get("output"), list):
        return "\n".join(item.get("content", "") for item in data["output"]).strip()

    raise RuntimeError("Unexpected Ollama response format.")

File: src/ollamaClient.js
import fetch from 'node-fetch';
import { config } from './config.js';

export async function generateAgentReply(messages, maxTokens = 250) {
  const endpoint = `${config.ollama.apiUrl}/v1/generate`;
  const body = {
    model: config.ollama.model,
    prompt: messages.map((message) => ({ role: message.role, content: message.content })),
    max_tokens: maxTokens,
    temperature: config.ollama.temperature,
  };

  const response = await fetch(endpoint, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(body),
  });

  if (!response.ok) {
    const errorText = await response.text();
    throw new Error(`Ollama API error ${response.status}: ${errorText}`);
  }

  const data = await response.json();
  if (typeof data.response === 'string') {
    return data.response.trim();
  }

  if (Array.isArray(data.output) && data.output.length > 0) {
    return data.output.map((item) => item.content).join('\n').trim();
  }

  throw new Error('Unexpected Ollama response format.');
}
